test/integrations/testTypes.ts

```diff
@@ -44,3 +45,98 @@ export type MockHttpCallsData = {
   httpReq: Record<string, any>;
   httpRes: Partial<AxiosResponse>;
 };
+
+export const arrayOrSingleElement = (schema: ZodTypeAny) => z.union([z.array(schema), schema]);
+export const formElementSchema = arrayOrSingleElement(
+  z.union([z.string(), z.number(), z.boolean()]),
+);
+export const eventResponseSchema = z.union([
```

Comment:
Need to enforce either success or error event related fields for each response - how will that be achieved if all are mentioned as optional ?

Reply:
currently, starting with only types validations, once this version is released, we can tighten it more